app/presenters/concerns/facet/url.rb

```diff
@@ -0,0 +1,97 @@
+# frozen_string_literal: true
+
+module Facet
+  module URL
+    extend ActiveSupport::Concern
+
+    ##
+    # URL for a facet item to link to
+    #
+    # If the facet item is already selected, this URL will remove it. If not, it
+    # will add it.
+    #
+    # @param see {#facet_item}
+    # @return [String] URL to add/remove the facet item from the search
+    def facet_item_url(item)
+      if facet_config.single
+        replace_facet_url(item)
+      else
+        facet_in_params?(facet_name, item) ? remove_facet_url(item) : add_facet_url(item)
+      end
+    end
+
+    def add_facet_url(item)
+      rewrite_facet_url(:add, item)
+    end
+
+    def remove_facet_url(item)
+      rewrite_facet_url(:remove, item)
+    end
+
+    def replace_facet_url(item)
+      rewrite_facet_url(:replace, item)
+    end
+
+    def rewrite_facet_url(action, item)
+      path = send(:"#{action}_facet_path", item)
```

Comment:
all these actions are the same, no? Why not just call search_action_path here?

Reply:
So that sub-classes may override one or more of them.